depthai_bridge/include/depthai_bridge/ImuConverter.hpp

```diff
@@ -19,27 +22,187 @@ namespace ros {
 namespace ImuMsgs = sensor_msgs::msg;
 using ImuPtr = ImuMsgs::Imu::SharedPtr;
 
-enum class ImuSyncMethod { COPY, LINEAR_INTERPOLATE_GYRO, LINEAR_INTERPOLATE_ACCEL };
+enum class ImuSyncMethod { COPY, LINEAR_INTERPOLATE_GYRO, LINEAR_INTERPOLATE_ACCEL, LINEAR_INTERPOLATE_ROTATION };
 
 class ImuConverter {
    public:
     ImuConverter(const std::string& frameName,
                  ImuSyncMethod syncMode = ImuSyncMethod::LINEAR_INTERPOLATE_ACCEL,
                  double linear_accel_cov = 0.0,
-                 double angular_velocity_cov = 0.0);
+                 double angular_velocity_cov = 0.0,
+                 double rotation_cov = 0.0,
+                 double magnetic_field_cov = 0.0,
+                 bool enable_rotation = false);
     ~ImuConverter();
-    void toRosMsg(std::shared_ptr<dai::IMUData> inData, std::deque<ImuMsgs::Imu>& outImuMsg);
+    void toRosMsg(std::shared_ptr<dai::IMUData> inData, std::deque<ImuMsgs::Imu>& outImuMsgs);
+    void toRosDaiMsg(std::shared_ptr<dai::IMUData> inData, std::deque<depthai_ros_msgs::msg::ImuWithMagneticField>& outImuMsgs);
+
+    template <typename T>
+    T lerp(const T& a, const T& b, const double t) {
+        return a * (1.0 - t) + b * t;
+    }
+
+    template <typename T>
+    T lerpImu(const T& a, const T& b, const double t) {
+        T res;
+        res.x = lerp(a.x, b.x, t);
+        res.y = lerp(a.y, b.y, t);
+        res.z = lerp(a.z, b.z, t);
+        return res;
+    }
+
+    dai::IMUReportRotationVectorWAcc lerpImu(const dai::IMUReportRotationVectorWAcc& a, const dai::IMUReportRotationVectorWAcc& b, const double t) {
+        dai::IMUReportRotationVectorWAcc res;
+        res.i = lerp(a.i, b.i, t);
+        res.j = lerp(a.j, b.j, t);
+        res.k = lerp(a.k, b.k, t);
+        res.real = lerp(a.real, b.real, t);
```

Comment:
are you sure on this one ? We might need to do [slerp](https://splines.readthedocs.io/en/latest/rotation/slerp.html)
and also make sure the result is also unit sphere ?

Reply:
Ah, yes, forgot this one needs to be treated separately, though after rethinking that, I'd remove rotation interpolation and add it later if needed.